pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -407,6 +407,18 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		return reconcile.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// requeue if allowed disruptions in the default PDB is 0
+	allowedDisruptions, err := r.getAllowedDisruptions(osdPDBAppName, request.Namespace)
+	if err != nil {
+		logger.Debugf("failed to get allowed disruptions count from default osd pdb %q. Skipping reconcile. Error: %v", osdPDBAppName, err)
+		return reconcile.Result{}, nil
+	}
+
+	if allowedDisruptions == 0 {
+		logger.Info("reconciling osd pdb reconciler as the allowed disruptions in default pdb is 0")
```

Comment:
If an OSD is crashlooping, but the PGs are healthy because they have been backfilled, this will cause the pdb reconcile to continue every 30s until the OSD stops crashing, right? Or will this be skipped if the PGs are healthy?

Reply:
When OSDs are crashlooping and PGs are not healthy, then we won't be seeing the default PDB (osdPDBAppName). Only blocking PDBs would be there. This condition won't hit because of the error in line 412

Its a good point though. I'll test it a bit more and confirm.

Reply:
So OSD goes into CLBO and pgs become degraded. When pgs become active again after re-balancing, we end up with following: 
```
Every 2.0s: oc get pdb -n rook-ceph                                     localhost.localdomain: Tue Sep 14 12:04:08 2021

NAME                MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
rook-ceph-mon-pdb   N/A             1                 1                     37m
rook-ceph-osd       N/A             1                 0                     112s
```
Observe `Allowed Disruptions` is 0 because the OSD is still down.  PDB reconciles with [here](https://github.com/rook/rook/blob/ab728e0183c92e059af7d663b287b00e95d6e175/pkg/operator/ceph/disruption/clusterdisruption/osd.go#L350) 

Now user purges the OSD for the CLBO OSD pod.  PBDs move back to the desired state again 

```
$ oc get pdb -n rook-ceph 
NAME                MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
rook-ceph-mon-pdb   N/A             1                 1                     71m
rook-ceph-osd       N/A             1                 1                     36m
```

Reply:
Ok, so the only way to return to the state of clean OSD PVCs is to remove/replace the crashing OSD. And from your link, the PDBs will continue reconciling every 15s? That might be worth looking into for a separate PR so it isn't so frequent.

Reply:
Yes, thats true. Only way is to remove the crashed OSD. This behavior would still have been the same even before this PR. I'll a take a look (in a separate PR) if we handle this situation better.

Reply:
@sp98 please open up an issue to track this. Thanks